components/gitpod-db/src/typeorm/synchronizer.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Deferred } from "@gitpod/gitpod-protocol/lib/util/deferred";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { inject, injectable } from "inversify";
+import { TypeORM } from "./typeorm";
+
+@injectable()
+export class Synchronizer {
```

Comment:
What are we trying to achieve here? Find a way to do cluster-level synchronization? Because due to multiple DBs, we this won't get us any further, no? :thinking: 
And if we only need cluster-level, we could use RabbitMQConsensusLeaderMessenger instead?

Reply:
The synchronization would ideally be application (i.e. global) level. For the time until we have removed db-sync it won't obviously, but that is ok for the use case I have. I don't know how I would use `RabbitMQConsensusLeaderMessenger` for this. Also didn't we want to get rid of rabbitMQ at some point?

Reply:
Ok, given the context (cluster level ok, global desired) this make sense.

> I don't know how I would use RabbitMQConsensusLeaderMessenger for this

You are right: due to the on-request nature it does not make sense to look into non-DB mechanisms.

> Also didn't we want to get rid of rabbitMQ at some point?

I think everyone is waiting for the final nudge to replace it with sth different. But as it does not hurt in any (obvious) way nobody felt obliged so far to do it.